src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -341,7 +343,14 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     }      // for(orderedSubgraphs)
 
     std::map<std::size_t, std::string> forced_sub_devices{};
-    const std::string fsd_opt = m_cfg.get<::intel_npu::NPUW_SUBMODEL_DEVICE>();
+    std::string fsd_opt = m_cfg.get<::intel_npu::NPUW_SUBMODEL_DEVICE>();
+    // Change "last" keyword to tail subgraph number
+    std::size_t last_pos = fsd_opt.find("last");
+    if (last_pos != std::string::npos) {
+        fsd_opt.erase(last_pos, 4);
+        fsd_opt.insert(last_pos, std::to_string(last_sub_idx));
+    }
+
     forced_sub_devices = ::intel_npu ::OptionParser<std::map<std::size_t, std::string>>::parse(fsd_opt);
```

Comment:
@AsyaPronina is this `OptionParser` ours? Can we add extra parameters to `parse` in this case? I'd prefer that code handle `last` for all its users (assuming the end of the range is passed to `parse()`).

Reply:
This is NPU `OptionParser`, it is generic callable to parse from string to requested type. However, we can as it to parse to `std::map<std::string, std::string>` instead of `std::map<std::size_t, std::string>`, where key will be string, so it may accept "last". But then we need to parse other indices from `string` to `std::size_t` on our own.

Reply:
> This is NPU `OptionParser`, it is generic callable to parse from string to requested type. However, we can as it to parse to `std::map<std::string, std::string>` instead of `std::map<std::size_t, std::string>`, where key will be string, so it may accept "last". But then we need to parse other indices from `string` to `std::size_t` on our own.

so it means we can also extend it with a new `parse()` overload for this particular template instantiation, isn't it?

Reply:
Yes, sure, we can, but we should be careful to ensure that only we will have that particular instantiation

Reply:
Yes, you are right, thank you!